atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionManager.java

```diff
@@ -102,4 +102,9 @@
      * @return value returned by task
      */
     <T, E extends Exception> T runTaskReadOnly(TransactionTask<T, E> task) throws E;
+
+    /**
+     * Performs non-destructive cleanup when the transaction manager is no longer needed.
+     */
+    void close();
```

Comment:
Adding `isClose` and the state checks might break products that might perform some cleanup operations after the close. Also, this is best used in a try-with-resources block. So, I think that the sanity check is not super-important here.